test/unit/auth/auth.spec.ts

```diff
@@ -1401,10 +1407,204 @@ AUTH_CONFIGS.forEach((testConfig) => {
           })
           .catch((error) => {
             expect(error).to.have.property('code', 'auth/argument-error');
-            expect(validator.isNonNullObject).to.have.been.calledOnce.and.calledWith(null);
+            expect(validator.isNonNullObject).to.have.been.calledWith(null);
+          });
+      });
+
+      const invalidUpdateRequests: UpdateRequest[] = [
+        { providerToLink: { uid: 'google_uid' } },
+        { providerToLink: { providerId: 'google.com' } },
+        { providerToLink: { providerId: 'google.com', uid: '' } },
+        { providerToLink: { providerId: '', uid: 'google_uid' } },
+      ];
+      invalidUpdateRequests.forEach((invalidUpdateRequest) => {
+        it('should be rejected given an UpdateRequest with an invalid providerToLink parameter', () => {
+          expect(() => {
+            auth.updateUser(uid, invalidUpdateRequest);
+          }).to.throw(FirebaseAuthError).with.property('code', 'auth/argument-error');
+        });
+      });
+
+      it('should rename providerToLink property to linkProviderUserInfo', async () => {
+        const invokeRequestHandlerStub = sinon.stub(testConfig.RequestHandler.prototype, 'invokeRequestHandler')
+          .resolves({
+            localId: uid,
+          });
+
+        // Stub getAccountInfoByUid to return a valid result (unchecked; we
+        // just need it to be valid so as to not crash.)
+        const getUserStub = sinon.stub(testConfig.RequestHandler.prototype, 'getAccountInfoByUid')
+          .resolves(expectedGetAccountInfoResult);
+
+        stubs.push(invokeRequestHandlerStub);
+        stubs.push(getUserStub);
+
+        await auth.updateUser(uid, {
+          providerToLink: {
+            providerId: 'google.com',
+            uid: 'google_uid',
+          },
+        });
+
+        expect(invokeRequestHandlerStub).to.have.been.calledOnce.and.calledWith(
+          sinon.match.any, sinon.match.any, {
+            localId: uid,
+            linkProviderUserInfo: {
+              providerId: 'google.com',
+              rawId: 'google_uid',
+            },
+          });
+      });
+
+      it('should be rejected given an empty providersToDelete list', () => {
+        expect(() => {
+          auth.updateUser(uid, {
+            providersToDelete: [],
```

Comment:
Could this be a no-op instead?

Reply:
You're right; there's not really any reason to reject this. I've moved the test to an integration test (to ensure the backend is happy with the noop. If that ever changes, we can add code to special case this.)